iothub/device/src/IotHubModuleClient.cs

```diff
@@ -138,6 +138,21 @@ public int DiagnosticSamplingPercentage
         public void SetConnectionStateChangeHandler(Action<ConnectionState, ConnectionStateChangeReason> stateChangeHandler)
             => InternalClient.SetConnectionStateChangeHandler(stateChangeHandler);
 
+        /// <summary>
+        /// Last connection state.
+        /// </summary>
+        public ConnectionState LastConnectionState => InternalClient._lastConnectionState.State;
```

Comment:
Store as one property: ConnectionInfo. Set it at the same time so no timing issues if someone checked between assignment of the 3 properties.